app/Console/Commands/SendWeeklyReport.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Notifications\WeeklyReportNotification;
+use App\Repositories\User\UserRepoInterface;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Notification;
+
+class SendWeeklyReport extends Command
+{
+    protected $userRepo;
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'report:weekly';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Send weekly report for admin by email';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct(UserRepoInterface $userRepo)
+    {
+        parent::__construct();
+
+        $this->userRepo = $userRepo;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $admins = $this->userRepo->getAllAdminAccounts();
+        $numberOfNewUsers = $this->userRepo->countNewUsersByWeek();
+
+        $admins->each(function ($admin) use ($numberOfNewUsers) {
+            Notification::send(
+                $admin,
+                new WeeklyReportNotification($admin->full_name, $numberOfNewUsers)
```

Comment:
sao send email mà em lại dùng noti nhờ? sao ko dùng luôn mailer

Reply:
Tại em muốn tận dụng luôn email template trong method toMail() của Notifcation ạ
`
/**
     * Get the mail representation of the notification.
     *
     * @param  mixed  $notifiable
     * @return \Illuminate\Notifications\Messages\MailMessage
     */
    public function toMail($notifiable)
    {
        return (new MailMessage)
            ->subject('[' . env('APP_NAME') . '] ' . 'Weekly report')
            ->line('This mail sent from MYMUSIC application to ' . $this->adminName . '.')
            ->line('Here is the announcement of the number of users for the past week.')
            ->line('The number of new users: ' . $this->numberOfNewUsers . '.')
            ->action('Click here to checkout', route('admin.users.index'))
            ->line('End!');
    }
`

Reply:
Và dùng Notification có thể thay đổi pphuowng thức thông báo, VD từ Mail sang Slack hoặc những channel khác